### PR TITLE
feat(cli): rotate file logs for coderd 

### LINKED ISSUE #15309: Rotate server logs
We currently [rotate agent logs](https://github.com/coder/coder/blob/330acd127071eaf0aec50cd72d51abffdff98bbb/cli/agent.go#L113-L118) and had a customer request to rotate the server logs as well for `CODER_LOGGING_HUMAN`

### PR DESCRIPTION
**Pull request by defelmnq**
Related to #15309 

As we already are doing for agent logs - this PR is enabling the logs rotation for coderd logs. 

Currently keeping the same logic than we had for agent - with 5MB as the file size for rotation.

### CODE REVIEW

File: cli/clilog/clilog.go [hunk 1]
```diff
@@ -112,13 +112,17 @@ func (b *Builder) Build(inv *serpent.Invocation) (log slog.Logger, closeLog func
...
-				return xerrors.Errorf("open log file %q: %w", loc, err)
+			logWriter := &lumberjack.Logger{
+				Filename: loc,
+				MaxSize:  5, // MB
+				// Without this, rotated logs will never be deleted.
+				MaxBackups: 1,
 			}
-			closers = append(closers, fi.Close)
-			sinks = append(sinks, sinkFn(fi))
+			closers = append(closers, logWriter.Close)
```
Comment (dannykopping) [inline cli/clilog/clilog.go:121]: Does this give us the equivalent behaviour of `lumberjackWriteCloseFixer` in `cli/agent.go`?

**Review by dannykopping — COMMENTED** [review]
LGTM, but want to ensure we have like-for-like behaviour with the agent.

File: cli/clilog/clilog.go (hunk 1 above)
Comment (johnstcn) [reply cli/clilog/clilog.go:121]: I think we should probably re-use that same fix here, although the main reason for that seems to have been to reduce CI flakes due to racy writes after close. We probably won't be logging to a file in CI in the general case.

**Review by johnstcn — COMMENTED** [review]
See inline comment below; bonus point for a test in `./cli` but I'm not going to insist on it for this PR.

**Comment by dannykopping** [comment]
> See inline comment below; bonus point for a test in `./cli` but I'm not going to insist on it for this PR.

Agreed on a test; I was thinking about this over the weekend.
We rely on this lumberjack lib, which has its own tests, but if we ever wanted to swap it out for another we'd want the functionality to remain the same. We can at least test the functional aspect.

**Review by johnstcn — APPROVED** [review]
I smoke-tested this locally and noted our existing test coverage in `TestServer/Logging.*`. Testing Lumberjack's rotation is probably way too much overhead here.

If we wanted to add a test, we could extract `addSinkIfNotProvided()` in `cli/clilog` and test that on its own.

**Comment by dannykopping** [comment]
> I smoke-tested this locally and noted our existing test coverage in `TestServer/Logging.*`. Testing Lumberjack's rotation is probably way too much overhead here.

I trust your judgment. I'm happy to merge as-is.

**Pull request merged**